tests/modules/programs/awscli/awscli.nix

```
@@ -0,0 +1,42 @@
+{ config, lib, pkgs, ... }:
+
+{
+  programs = {
+    awscli = {
+      enable = true;
+      settings = {
+        default = {
+          output = "json";
+          region = "eu-west-3";
+        };
+      };
+      credentials = {
+        iam = {
+          credential_process = "pass show aws";
+        };
+      };
+      enableBashIntegration = true;
+      enableZshIntegration = true;
+    };
+    bash.enable = true;
+    zsh.enable = true;
+  };
+
```

To avoid unnecessary download of awscli2 in the test:
```suggestion

  test.stubs.awscli2 = { };
 
```